### ISSUE TITLE
Xournalapp Tablet Mode: Enhancing the Touchscreen Experience with Infinite Canvas and Floating Toolbar

### ISSUE THREAD

**Issue by volkansari** [xournalpp/xournalpp#5869]
### Is Your Feature Request Related to a Problem?

Tablet Mode to Xournalapp is related to improving the overall user experience, particularly for those using tablets and touchscreens. Currently, users may find the interface cluttered with too many visible tools, making it less efficient when working on a tablet where screen real estate is more limited. Additionally, without an infinite canvas option, users are constrained by page boundaries, which limits flexibility in note-taking and drawing, especially for larger projects.

### Feature Description

The proposed Tablet Mode for Xournal++ introduces several key features aimed at enhancing the usability of the application on tablets and touchscreens. These features include:

Infinite Canvas:
This feature allows users to work on a limitless canvas, eliminating the constraints of traditional page sizes. Users will be able to scroll and zoom in any direction, making it ideal for extensive note-taking, mind mapping, or large drawings. This flexibility enables seamless work on complex projects without the need to switch between multiple pages.

Hidden Menus with Floating Toolbar/Palette:
In Tablet Mode, the user interface is designed to be minimalist, with menus and toolbars hidden by default to maximize screen space for content creation. A floating toolbar or palette will appear only when needed, providing quick access to frequently used tools. This toolbar can be docked or left floating, depending on user preference, and will disappear after the tool is selected to maintain a clutter-free workspace.

Touch-Optimized Gestures:
The floating toolbar and menus will be accessible through intuitive touch gestures. Swiping from the edge of the screen can reveal menus or tools, ensuring that they remain easy to reach without interrupting the flow of writing or drawing.
These features are designed to provide a more streamlined, distraction-free experience for tablet users, allowing them to fully utilize the advantages of touchscreens while maintaining focus on their work.

### Alternative Solutions

Xournalapp Tablet Mode: Enhancing Your Experience with New Features
Exciting news for Xournalapp users! A new Tablet Mode is on the way, specifically designed to improve the note-taking, drawing, and document-editing experience for users with touchscreens and tablets. This mode introduces two key features: infinite canvas and hidden menus with a floating toolbar or palette.

Infinite Canvas Mode
One of the standout features of the new Tablet Mode is the infinite canvas option. With this feature, users will no longer be constrained by the edges of a standard page. You’ll be able to take notes, sketch, or create mind maps with no spatial limitations. The infinite canvas is particularly useful for large projects, creative workflows, and brainstorming sessions, offering limitless space to enhance productivity.

Key advantages:

Unlimited Space: Freely write and draw without worrying about running out of room on the page.
Zoom In/Out: Work at any level of detail, thanks to robust zooming and panning capabilities.
Extensive Content Management: Ideal for handling long notes and complex sketches, giving users more flexibility in organizing their work.
Hidden Menus and Floating Toolbar
Another major improvement in Tablet Mode is the introduction of hidden menus and a floating toolbar/palette. These features streamline the user interface, allowing you to dedicate more screen space to your content. The tools and menus will stay hidden until you need them, reducing distractions and letting you focus entirely on your work.

How does it work?

Floating Toolbar: Instead of a permanently fixed toolbar, a floating palette will slide in when needed, then disappear when you're done. This makes better use of screen space, especially on smaller devices.
Hidden Menus: Menus will be accessible via touch gestures from the edges of the screen, opening and closing as needed. This intuitive design ensures you have easy access to tools without cluttering the interface.
Optimized for Tablet Users
These new features are especially optimized for tablet users. Designed with touchscreens in mind, this mode will make it easier than ever to take handwritten notes, quickly sketch ideas, or edit documents on the go. The tablet mode also works seamlessly with stylus pens, providing a natural writing and drawing experience.

Main benefits:

Touchscreen-Friendly: Menus and toolbars can be accessed or hidden with simple touch gestures, providing an effortless workflow.
Maximized Screen Space: By hiding unnecessary tools and menus, more screen real estate is available for your content.
Customizable Experience: The floating toolbar and menus can be customized to your preferences, allowing you to tailor the interface to your working style.

**Comment by rolandlo** [comment]
Have you already discovered that the toolbars are customizable and that there is a floating toolbox that is customizable as well?

**Comment by volkansari** [comment]
Yes, I know that this can be done at least, but a user-friendly interface design would be more useful. I don't think the floating toolbox is user-friendly, I would like to give you a few sample screenshots on this subject.

![images (18).jpeg](https://github.com/user-attachments/assets/9134b407-0538-4126-b6d3-54942f8ebfd2)

![images (14).jpeg](https://github.com/user-attachments/assets/f813e3a9-a07b-4b8b-860c-4bfa1fc3fd40)

**Comment by volkansari** [comment]
Those who actively use these applications want the entire screen to be a blank page. Imagine that all the menus of your program are in a floating toolbox and the floating toolbox is minimized with popup menus and arrows. Imagine that the program is as functional as possible and at the same time simple. What you mean by simplicity is that there are fewer buttons on the screen.

**Comment by volkansari** [comment]
@bhennion 
@rolandlo 

Thank you for all your efforts and contributions to the free software world.
